app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -504,4 +313,69 @@ public void onServiceDisconnected(ComponentName arg0) {
             mBound = false;
         }
     };
+
+    private String findSmsData(String[] prebuffer,String[] postbuffer){
```

Comment:
I also seperated number and sms into one function for easier debugging and new devs to find things easier